### ISSUE TITLE
Windows status

### ISSUE THREAD

**Issue by alonbl** [phaethon/scapy#15]
Hi,

I am opening this as there is no other method to communicate with you. You are doing a great work, I see you integrated the winpcapy, but I could not actually use it even in most basic method, iface is not initialized if I workaround it I get complain that pcap fd is not selectable etc...

Is there any method to use it even in primitive l2 only mode? I see windump can be used and all that is required is completion of network topology.

Thanks,

**Comment by phaethon** [comment]
Please, add more details.
Windows version
Interface type (wireless, or Ethernet)
What is your objective capture and dissect; or craft packets?

**Comment by alonbl** [comment]
Thanks for replying!

Windows version: Windows 8
Interface type: Ethernet (actually this is qemu guest)
Objective: capture and dissect all level and craft, however when sending packets for starters it will be great if at least sendp(packet, iface) will work, no need to have knowledge of topology, routing etc...

For now, I am using winpcapy and it works very well, so lower level should be ok.

**Comment by phaethon** [comment]
Presently, Windows is not the platform where development is focused. This is summary status of Windows support.
Packet dissecting (from pcap files):
Everything should work normally.
Packet crafting:
scapy < 3 uses libdnet to craft packets in Windows, but it is pain to install, and, especially, for python 3. winpcap (and winpcapy) has some limited support for sending packets. It is not well integrated in scapy3k yet, though it is planned. You can use send method of _Pcap_Wrapper_pypcap class. Any help on further developing this is welcome.
Packet sniffing:
Sniffing should work as is (assuming winpcap is working), but issues with wireless interface sniffing on Windows is a deal breaker for many people. Other tools also have major issues with wireless interface support. Any ideas how to move forward on this are welcome.

**Comment by alonbl** [comment]
Thank you for the update. Maybe if I will have some time I will help to port libdnet.

**Comment by phaethon** [comment]
Libdnet development seems to have stopped, and so is its availability on Windows platform. Looking ahead, I see winpcap as a better alternative as it is MUCH easier to install and is well supported. If you are willing to invest time, checking out winpcap packet sending functionality, its limitations if any, and possibly refactoring scapy code to use it. 
btw, packet sending is unique to winpcap, as libpcap does not have such capability.

**Comment by alonbl** [comment]
I do not understand.... I use winpcapy on both linux and windows and can send and receive packets. In linux it is using libpcap.

**Comment by phaethon** [comment]
Sorry, for confusion. I deleted that comment, which seems to be obsolete. There used to be different implementations of packet sending in winpcap and libpcap, but both of them can be used to send packet.

**Comment by phaethon** [comment]
I did some testing today and can confirm that packet analysis and offline crafting (i.e. rdpcap, wrpcap, layers, etc., but not sending, sniffing, receiving) is working on Windows 10 and Windows 7 using Anaconda python 3.5 install. Also, winpcap and powershell is required. Netifaces module is optional.

I have started additional development to make sniffing work. Anybody interested in testing can install latest git version and check value of `ifaces` variable. It should contain network interfaces, ip, mac addresses, winpcap name of the interface.
